bpf/kmesh/ads/include/local_ratelimit.h

```diff
@@ -0,0 +1,201 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Kmesh */
+
+#ifndef __KMESH_LOCAL_RATE_LIMIT_H__
+#define __KMESH_LOCAL_RATE_LIMIT_H__
+
+#include "bpf_log.h"
+#include "bpf_common.h"
+#include "kmesh_common.h"
+#include "listener/listener.pb-c.h"
+
+struct ratelimit_key {
+    union {
```

Comment:
why use union here

Reply:
Using a union in the ratelimit_key structure improves scalability. If we need to limit HTTP connections in the future, we can reuse the same key structure. For example, we can add a structure for HTTP connections like this:

```

struct ratelimit_key {
    union {
        struct {
            __u64 netns;  /* Network namespace. */
            __u32 ipv4;   /* Destination IPv4 address. */
            __u32 port;   /* Destination port. */
            __u32 family; /* Address family (e.g., AF_INET) */
        } sk_skb;

        struct {
            __u32 http_method;   /* HTTP method (e.g., GET, POST) */
            __u32 http_version;    /* HTTP version (e.g., 1.1, 2) */
            __u32 http_port;         /* HTTP port */
        } http; // New HTTP connection related fields
    } key;
};

```